src/main/scala/io/iohk/ethereum/db/storage/BlockHeadersStorage.scala

```diff
@@ -24,7 +25,23 @@ class BlockHeadersStorage(val dataSource: DataSource) extends KeyValueStorage[Bl
   override def valueDeserializer: (IndexedSeq[Byte]) => BlockHeader =
     (encodedBlockHeader: IndexedSeq[Byte]) => rlpDecode[BlockHeader](encodedBlockHeader.toArray)
 
-  override protected def apply(dataSource: DataSource): BlockHeadersStorage = new BlockHeadersStorage(dataSource)
+  override protected def apply(dataSource: DataSource): BlockHeadersStorage =
+    new BlockHeadersStorage(dataSource, blockHeadersNumbersStorage)
+
+  def get(blockNumber: BigInt): Option[BlockHeader] = if (blockNumber == 0) {
+    //todo get block from config, or we will put 0 block to storage at app startup if it is not there?
+    None
+  } else {
+    blockHeadersNumbersStorage.get(blockNumber).flatMap(key => super.get(key))
+  }
+
+  override def update(toRemove: Seq[BlockHeaderHash], toUpsert: Seq[(BlockHeaderHash, BlockHeader)]): BlockHeadersStorage = {
```

Comment:
great, missed that, so you added another layer (repository) instead of using storages directly, look good to me.
Is there API to get blocks by number?